oshi-core/src/main/java/oshi/util/FileUtil.java

```diff
@@ -217,40 +221,56 @@ public static String getStringFromFile(String filename) {
     }
 
     /**
-     * Read a configuration file from the class path and return its properties
+     * Read a configuration file from the sequence of context classloader, system
+     * classloader and classloader of the current class, and return its properties
      *
      * @param propsFilename
      *            The filename
      * @return A {@link Properties} object containing the properties.
      */
     public static Properties readPropertiesFromFilename(String propsFilename) {
         Properties archProps = new Properties();
-        // Load the configuration file from the classpath
-        try {
-            ClassLoader loader = Thread.currentThread().getContextClassLoader();
-            if (loader == null) {
-                loader = ClassLoader.getSystemClassLoader();
-                if (loader == null) {
-                    throw new IOException();
-                }
+        // Load the configuration file from the different classloaders
+        Set<ClassLoader> loaders = Stream.of(
```

Comment:
This whole sequence seems overly complicated for what's essentially three if/then checks.  

Why not something like:
```
    if (readPropertiesFromClassLoader(propsFilename, archProps,
            Thread.currentThread().getContextClassLoader())
    || readPropertiesFromClassLoader(propsFilename, archProps,
            ClassLoader.getSystemClassLoader())
    || readPropertiesFromClassLoader(propsFilename, archProps,
            FileUtil.class.getClassLoader())) {
        return archProps;
    }
```

Reply:
when I was testing, Thread.currentThread().getContextClassLoader() == ClassLoader.getSystemClassLoader() but perhaps it's early optimization for that

Reply:
If you'd prefer to reorder them (FileUtil class 2nd and System 3rd?) that's fine with me.

Reply:
In most of the application I think all 3 classloaders are the same, so it shouldn't matter too much
`FileUtil.class.getClassLoader()` should always gets the default properties as it is the classloader loading the oshi-core.jar unless the jar is being altered / shaded, keeping it as the last one seems better for others need to override the properties

Reply:
OK, great!  I'll go ahead and merge this!

Reply:
Almost a year later, I see why you did this and wish I'd left your initial contribution alone. :)